src/to_mathlib/analysis/inner_product_space/tensor_product.lean

```diff
@@ -28,118 +87,12 @@ linear_map.mul' ℝ ℝ ∘ₗ
   tensor_product_aux E F ((e₁ ⊗ₜ[ℝ] f₁) ⊗ₜ[ℝ] (e₂ ⊗ₜ[ℝ] f₂)) = ⟪e₁, e₂⟫ * ⟪f₁, f₂⟫ :=
 by simp [tensor_product_aux, as_tensor]
 
-/- We may not actually need this for this project since what we really want is the `instance`
-lower down for the _tensor power_.
-
-This is still a gap in Mathlib that is worth filling. -/
-
--- new stuff starts here (line 30 in my analysis/inner_product_space/tensor_product file)
-
-
-section some_linear_algebra_from_kevin
-
-local attribute [ext] tensor_product.ext
-
-/-- Two linear maps (M ⊗ N) ⊗ (P ⊗ Q) → S which agree on all elements of the
-form (m ⊗ₜ n) ⊗ₜ (p ⊗ₜ q) are equal. -/
-theorem ext_fourfold' {R M N P Q S: Type*} [comm_semiring R]
-  [add_comm_monoid M] [add_comm_monoid N] [add_comm_monoid P] [add_comm_monoid Q]
-  [add_comm_monoid S] [module R M] [module R N] [module R P] [module R Q] [module R S]
-  {φ ψ : (M ⊗[R] N) ⊗[R] (P ⊗[R] Q) →ₗ[R] S}
-  (H : ∀ w x y z, φ ((w ⊗ₜ x) ⊗ₜ (y ⊗ₜ z)) = ψ ((w ⊗ₜ x) ⊗ₜ (y ⊗ₜ z))) : φ = ψ :=
-begin
-  ext m n p q,
-  exact H m n p q,
-end
-
-end some_linear_algebra_from_kevin
-
-/-- We may not actually need this for this project since what we really want is the `instance`
-lower down for the _tensor power_.
+-- instance (k : ℕ) : inner_product_space ℝ (⨂[ℝ]^k E) := sorry
 
-This is still a gap in Mathlib that is worth filling. -/
---variables (hE : finite_dimensional ℝ E) (hF : finite_dimensional ℝ F)
---include hE hF
-instance : inner_product_space ℝ (E ⊗[ℝ] F) := of_core
-{ inner := λ x y, tensor_product_aux E F (x ⊗ₜ y),
-  conj_sym :=
-  begin
-    sorry {
-    intros x y,
-    rw [is_R_or_C.conj_to_real],
-    suffices : tensor_product_aux E F =
-      (tensor_product_aux E F).comp (tensor_product.comm _ _ _).to_linear_map,
-    { conv_lhs {rw this},
-      simp, },
-    apply ext_fourfold',
-    intros e₁ f₁ e₂ f₂,
-    suffices : inner e₁ e₂ * inner f₁ f₂ = inner e₂ e₁ * inner f₂ f₁,
-    { simpa },
-    rw [← inner_conj_sym e₁, is_R_or_C.conj_to_real, ← inner_conj_sym f₁, is_R_or_C.conj_to_real],}
-  end,
-  nonneg_re := begin
-    intros z,
-    rw [is_R_or_C.re_to_real],
-    rw [← (basis.of_vector_space ℝ (E ⊗ F)).total_repr z, basis.coe_of_vector_space],
-    rw finsupp.total_apply,
-    sorry
-  end,
-  definite := λ x, begin
-    sorry
-  end,
-  add_left := λ x y z, by rw [tensor_product.add_tmul, map_add],
-  smul_left := λ x y r, by rw [is_R_or_C.conj_to_real,
-    ← tensor_product.smul_tmul', map_smul, smul_eq_mul],}
--- omit hE hF
+-- @[simp] lemma inner_tpow {k : ℕ} (e₁ e₂ : fin k → E) :
+--   ⟪tensor_power.tpow ℝ e₁, tensor_power.tpow ℝ e₂⟫ = ∏ i, ⟪e₁ i, e₂ i⟫ :=
+-- sorry
```

Comment:
Likewise we need to keep this.
```suggestion
@[simp] lemma inner_tpow {k : ℕ} (e₁ e₂ : fin k → E) :
  ⟪tensor_power.tpow ℝ e₁, tensor_power.tpow ℝ e₂⟫ = ∏ i, ⟪e₁ i, e₂ i⟫ :=
sorry
```

Reply:
re-added

Reply:
As above: we don't want to keep commented-out code, especially code that exists elsewhere.